src/stactools/package/stac.py

```diff
@@ -0,0 +1,179 @@
+from datetime import datetime
+import logging
+
+from shapely.geometry import Polygon
+from pystac.extensions.projection import ProjectionExtension
+from pystac import (
+    CatalogType,
+    Collection,
+    Item,
+    Asset,
+    Extent,
+    SpatialExtent,
+    TemporalExtent,
+    Link,
+    Provider,
+    MediaType,
+)
+
+logger = logging.getLogger(__name__)
+
+
+# This includes a set of minimum requirements for the creation of a STAC Collection,
+# and should be added to/modified as necessary to develop a dataset-specific package.
+def create_collection(
+    title: str,
+    description: str,
+    catalog_type: str,
+    providers: list[Provider],
+    license: str,
+    license_link: Link,
+    extent: list,
+    start_time: datetime,
+    end_time: datetime,
+    href: str = None,
+) -> Collection:
```

Comment:
So this pretty closely mirrors `Collection.__init__`, so I'm wondering what benefit we get out of providing this function skeleton (anti-DRY)?

Reply:
Agreed. Any suggestions for a template? Something like this maybe:

```python
def create_collection(*args, **kwargs) -> Collection:
    """Create a STAC Collection

    See https://pystac.readthedocs.io/en/latest/api.html#collection

    Returns:
        Collection: STAC Collection object
    """
    collection = Collection(*args, **kwargs)

    return collection
```

Reply:
On second thought, the `NotImplementedError` route seems more intuitive to me. Something like this:

```python
def create_collection() -> Collection:
    """Create a STAC Collection

    Returns:
        Collection: STAC Collection object
    """
    # This function should include logic to extract all relevant metadata from
    # an asset describing the STAC collection and/or metadata coded into an
    # accompanying constants.py file
    
    # See https://pystac.readthedocs.io/en/latest/api.html#collection for requirements

    raise NotImplementedError()
```